models/executor.js

```diff
@@ -1,8 +1,14 @@
+var uuid = require('uuid');
+
 module.exports = function(Executor) {
-  Executor.beforeCreate = function(next) {
-    if (this.remainingCapacity === -1) {
+  Executor.definition.properties.id.default = function() {
+    return uuid.v4();
+  };
+
+  Executor.beforeSave = function(callback) {
+    if (this.remainingCapacity === null && this.totalCapacity !== null) {
```

Comment:
@rmg Moved it to beforeSave because the steps of registering an executor have changed. You must first create the executor and then start it with the generated ID and auth token. Once it starts, it updates the values.

Cant use `default` function since that needs to be independent and cannot be based on values of other fields in the model.